src/trade_ui.cpp

```diff
@@ -102,17 +100,20 @@ bool trade_preset::cat_sort_compare( const inventory_entry &lhs, const inventory
 }
 
 trade_ui::trade_ui( party_t &you, npc &trader, currency_t cost, std::string title )
-    : _upreset{ you, trader }, _tpreset{ trader, you },
-      _panes{ std::make_unique<pane_t>( this, trader, _tpreset, std::string(), _pane_size(),
-                                        _pane_orig( -1 ) ),
-              std::make_unique<pane_t>( this, you, _upreset, std::string(), _pane_size(),
-                                        _pane_orig( 1 ) ) },
+    : cataimgui::window( title, ImGuiWindowFlags_AlwaysAutoResize ),
+      _upreset{ you, trader }, _tpreset{ trader, you },
+      _panes { std::make_unique<pane_t>( this, trader, _tpreset, std::string() ),
+               std::make_unique<pane_t>( this, you, _upreset, std::string() ) },
       _parties{ &trader, &you }, _title( std::move( title ) )
 
 {
+    window_size.x = 0;
+    window_size.y = 0;
     _panes[_you]->add_character_items( you );
     _panes[_you]->add_nearby_items( 1 );
     _panes[_trader]->add_character_items( trader );
+    _panes[_trader]->set_title( "Trader Inventory" ); // DO NOT REMOVE: ImGui needs each child window to have unique titles/IDs
+    _panes[_you]->set_title( "Your Inventory" );
```

Comment:
Are these two strings displayed in GUI? If so they need to be localised:

```suggestion
    _panes[_trader]->set_title( _( "Trader Inventory" ) ); // DO NOT REMOVE: ImGui needs each child window to have unique titles/IDs
    _panes[_you]->set_title( _( "Your Inventory" ) );
```

Reply:
Yes, they certainly are, sounds good.